src/app/Providers/components/VMwareProviderHostsTable/SelectNetworkModal.tsx

```diff
@@ -10,63 +10,85 @@ import {
+const useSelectNetworkFormState = (
+  selectedHosts: IHost[],
+  hostConfigs: IHostConfig[],
+  provider: IVMwareProvider
+) => {
+  const existingHostConfigs = getExistingHostConfigs(selectedHosts, hostConfigs, provider);
+  const existingIpAddresses = existingHostConfigs
+    .map((config) => config?.spec.ipAddress)
+    .filter((ip) => !!ip) as string[];
+  const allOnSameIp = Array.from(new Set(existingIpAddresses)).length === 1;
+  const preselectedAdapter = allOnSameIp
+    ? selectedHosts[0].networkAdapters.find(
+        (adapter) => adapter.ipAddress === existingIpAddresses[0]
+      ) || null
+    : null;
+
+  const isReusingCredentials = useFormField(true, yup.boolean().required());
+  const usernameSchema = yup.string().max(320).label('Host admin username');
+  const passwordSchema = yup.string().max(256).label('Host admin password');
+  return useFormState({
+    selectedNetworkAdapter: useFormField<IHostNetworkAdapter | null>(
+      preselectedAdapter,
+      yup.mixed<IHostNetworkAdapter>().label('Host network').required()
+    ),
+    adminUsername: useFormField(
+      '',
+      isReusingCredentials.value ? usernameSchema : usernameSchema.required()
+    ),
+    adminPassword: useFormField(
+      '',
+      isReusingCredentials.value ? passwordSchema : passwordSchema.required()
+    ),
+    isReusingCredentials,
+  });
+};
+
+export type SelectNetworkFormValues = ReturnType<typeof useSelectNetworkFormState>['values'];
 
 const SelectNetworkModal: React.FunctionComponent<ISelectNetworkModalProps> = ({
   selectedHosts,
+  hostConfigs,
+  provider,
   onClose,
 }: ISelectNetworkModalProps) => {
-  const form = useFormState({
-    adminUserId: useFormField('', yup.string().max(320).label('User Id').required()),
-    adminPassword: useFormField('', yup.string().max(256).label('Password').required()),
-    selectedNetwork: useFormField(null, yup.mixed<IHostNetwork>().label('Host network').required()),
-  });
+  const [configureHosts, configureHostsResult] = useConfigureHostsMutation(
+    provider,
+    selectedHosts,
+    hostConfigs,
+    onClose
+  );
 
-  // TODO Replace once https://github.com/konveyor/virt-ui/issues/85 has been addressed.
-  console.log('TODO: generate options for selected hosts: ', selectedHosts);
-  const networkOptions = MOCK_NETWORKS.map((network) => ({
-    toString: () => formatHostNetwork(network),
-    value: network.name,
-    props: { description: `${network.bandwidth}; MTU:${network.mtu}` },
-  }));
-  // End TODO
+  const form = useSelectNetworkFormState(selectedHosts, hostConfigs, provider);
 
-  const add = () => {
-    alert('TODO');
-    if (form.isValid) {
-      console.log('TODO: submit form!', form.values);
-      onClose();
-    }
-  };
+  const commonNetworkAdapters: IHostNetworkAdapter[] = selectedHosts[0].networkAdapters.filter(
+    ({ name, ipAddress }) =>
+      selectedHosts.every((host) =>
+        host.networkAdapters.some((na) => na.name === name && na.ipAddress === ipAddress)
```

Comment:
I think we need to match only the IP addresses for common network adapters..
Because if for some reason the operator did a typo or entered a different name to that interface then we won't have a match although VMs would be sharing the same Physical IP address.

Reply:
Per our discussion, we're using only IP addresses to match common adapters.